scio-parquet/src/main/scala/com/spotify/scio/parquet/read/ParquetRead.scala

```diff
@@ -43,8 +58,92 @@ object ParquetRead {
     readSupportFactory: ReadSupportFactory[T],
     conf: SerializableConfiguration,
     projectionFn: SerializableFunction[T, R]
-  ): ParDo.SingleOutput[ReadableFile, R] = {
+  ): PTransform[PCollection[ReadableFile], PCollection[R]] = {
     val sdf = new ParquetReadFn[T, R](readSupportFactory, conf, projectionFn)
-    ParDo.of(sdf)
+    val tfx: PTransform[PCollection[_ <: ReadableFile], PCollection[R]] = ParDo.of(sdf)
+
+    tfx.asInstanceOf[PTransform[PCollection[ReadableFile], PCollection[R]]]
+  }
+
+  def readTyped[T: ClassTag: Coder: ParquetType](
+    predicate: FilterPredicate = null,
+    conf: Configuration = null
+  ): PTransform[PCollection[ReadableFile], PCollection[T]] = readTyped[T, T](
+    identity,
+    predicate,
+    conf
+  )
+
+  def readTyped[T: ClassTag: Coder: ParquetType, R](
+    projectionFn: T => R,
+    predicate: FilterPredicate,
+    conf: Configuration
+  ): PTransform[PCollection[ReadableFile], PCollection[R]] = {
+    val configuration = ParquetConfiguration.ofNullable(conf)
+    Option(predicate).foreach(p => ParquetInputFormat.setFilterPredicate(configuration, p))
+
+    val cleanedFn = Functions.serializableFn(ClosureCleaner.clean(projectionFn))
+    readFiles(ReadSupportFactory.typed[T], new SerializableConfiguration(configuration), cleanedFn)
+  }
+
+  def readAvroGenericRecordFiles(
+    schema: Schema,
+    predicate: FilterPredicate = null,
+    conf: Configuration = null
+  ): PTransform[PCollection[ReadableFile], PCollection[GenericRecord]] =
+    readAvroGenericRecordFiles[GenericRecord](schema, identity, predicate, conf)
+
+  def readAvroGenericRecordFiles[T](
+    schema: Schema,
+    projectionFn: GenericRecord => T,
+    predicate: FilterPredicate,
+    conf: Configuration
+  ): PTransform[PCollection[ReadableFile], PCollection[T]] = {
+    val configuration = ParquetConfiguration.ofNullable(conf)
+    configuration.setBoolean(AvroReadSupport.AVRO_COMPATIBILITY, false)
+    AvroReadSupport.setAvroReadSchema(configuration, schema)
+    AvroReadSupport.setRequestedProjection(configuration, schema)
+
+    Option(predicate).foreach(p => ParquetInputFormat.setFilterPredicate(configuration, p))
+    if (configuration.get(AvroReadSupport.AVRO_DATA_SUPPLIER) == null) {
+      AvroReadSupport.setAvroDataSupplier(configuration, classOf[GenericDataSupplier])
+    }
+
+    val cleanedFn = Functions.serializableFn(ClosureCleaner.clean(projectionFn))
+    readFiles(
+      ReadSupportFactory.avro,
+      new SerializableConfiguration(configuration),
+      cleanedFn
+    )
+  }
+
+  def readAvro[T <: SpecificRecordBase: ClassTag](
```

Comment:
IMHO we should define API with weakest abstraction `SpecificRecord` instead. See #4459